x-pack/plugins/enterprise_search/public/applications/shared/layout/nav_link_helpers.ts

```diff
@@ -11,38 +11,45 @@ import { stripTrailingSlash } from '../../../../common/strip_slashes';
 
 import { KibanaLogic } from '../kibana';
 import { generateReactRouterProps, ReactRouterProps } from '../react_router_helpers';
+import { GeneratedReactRouterProps } from '../react_router_helpers/generate_react_router_props';
 
 interface Params {
   to: string;
-  isRoot?: boolean;
   shouldShowActiveForSubroutes?: boolean;
   items?: Array<EuiSideNavItemType<unknown>>; // Primarily passed if using `items` to determine isSelected - if not, you can just set `items` outside of this helper
 }
 
-export const generateNavLink = ({ to, items, ...rest }: Params & ReactRouterProps) => {
-  return {
-    ...generateReactRouterProps({ to, ...rest }),
-    isSelected: getNavLinkActive({ to, items, ...rest }),
-    items,
+type NavLinkProps = GeneratedReactRouterProps &
+  Pick<EuiSideNavItemType<unknown>, 'isSelected' | 'items'>;
+
+export const generateNavLink = ({ items, ...rest }: Params & ReactRouterProps): NavLinkProps => {
+  const linkProps = {
+    ...generateReactRouterProps({ ...rest }),
+    isSelected: getNavLinkActive({ items, ...rest }),
   };
+  return items ? { ...linkProps, items } : linkProps;
 };
 
 export const getNavLinkActive = ({
   to,
-  isRoot = false,
   shouldShowActiveForSubroutes = false,
   items = [],
-}: Params): boolean => {
+  shouldNotCreateHref = false,
+}: Params & ReactRouterProps): boolean => {
   const { pathname } = KibanaLogic.values.history.location;
+  // @ts-ignore TODO: Not sure where history.basePath is coming from. Is Kibana inserting it?
```

Comment:
suggestion: use some typecasting instead of ts-ignore, or maybe it's better to just change the typing in KibanaLogic to include basePath